### PR TITLE
Initial Server Setup

### PR DESCRIPTION
**Pull request by joyguptaa**
Date: 5 Oc 2024

Developer Name: Joy Gupta

---

## Issue Ticket Number
- https://github.com/Real-Dev-Squad/discord-service/issues/1

## Description

This PR contains code for the initial setup of the discord service. 

### Documentation Updated?

- [ ] Yes
- [x] No, will raise a separate PR for that

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots

https://github.com/user-attachments/assets/e8df5fdc-0956-4aa1-a9ca-745362269d43

### CODE REVIEW

**Comment by iAmAshuSahoo** [comment]
LGTM

**Review by iAmAshuSahoo — APPROVED** [review]
LGTM

**Review by iamitprakash — CHANGES_REQUESTED** [review]
please change PR title

**Comment by joyguptaa** [comment]
> please change PR title

done

File: controllers/baseHandler.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/julienschmidt/httprouter"
+	"github.com/sirupsen/logrus"
+)
+
+func HomeHandler(rw http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	rw.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	payload, err := io.ReadAll(r.Body)
+	if err != nil {
+		logrus.Error(err)
+		http.Error(rw, "Failed to read request body", http.StatusBadRequest)
```
Comment (iamitprakash) [inline controllers/baseHandler.go:19]: please return after line #19

File: controllers/baseHandler.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	}
+	var message dtos.DiscordMessage
+	err = json.Unmarshal(payload, &message)
+	if err != nil {
+		logrus.Error(err)
+		http.Error(rw, "Failed to read request body", http.StatusBadRequest)
+	}
+	switch message.Type {
+	case discordgo.InteractionPing:
+		rw.WriteHeader(http.StatusOK)
```
Comment (iamitprakash) [inline controllers/baseHandler.go:29]: remove this line

File: controllers/baseHandler.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		resp := map[string]uint8{"type": uint8(discordgo.InteractionResponsePong)}
+		err = json.NewEncoder(rw).Encode(resp)
+		if err != nil {
+			rw.WriteHeader(http.StatusInternalServerError)
+			http.Error(rw, "Internal Server Error", http.StatusInternalServerError)
+			logrus.Error(err)
+		}
+		return
+	default:
+		rw.WriteHeader(http.StatusOK)
```
Comment (iamitprakash) [inline controllers/baseHandler.go:39]: remove line #39

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: controllers/baseHandler.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	"io"
+	"net/http"
+
+	"github.com/Real-Dev-Squad/discord-service/dtos"
+	"github.com/bwmarrin/discordgo"
+	"github.com/julienschmidt/httprouter"
+	"github.com/sirupsen/logrus"
+)
+
+func HomeHandler(rw http.ResponseWriter, r *http.Request, p httprouter.Params) {
```
Comment (iamitprakash) [inline controllers/baseHandler.go:14]:
```
func HomeHandler(rw http.ResponseWriter, r *http.Request, p httprouter.Params) {
	rw.Header().Set("Content-Type", "application/json; charset=UTF-8")
	
	payload, err := io.ReadAll(r.Body)
	if err != nil {
		logrus.Error(err)
		http.Error(rw, "Failed to read request body", http.StatusBadRequest)
		return
	}
	
	var message dtos.DiscordMessage
	err = json.Unmarshal(payload, &message)
	if err != nil {
		logrus.Error(err)
		http.Error(rw, "Failed to parse request body", http.StatusBadRequest)
		return
	}
	
	switch message.Type {
	case discordgo.InteractionPing:
		resp := map[string]uint8{"type": uint8(discordgo.InteractionResponsePong)}
		err = json.NewEncoder(rw).Encode(resp)
		if err != nil {
			logrus.Error(err)
			http.Error(rw, "Internal Server Error", http.StatusInternalServerError)
			return
		}
	default:
		
		rw.WriteHeader(http.StatusOK)
	}
}

```

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: controllers/baseHandler.go (hunk 4 above)
Comment (joyguptaa) [reply controllers/baseHandler.go:14]: done

**Pull request merged**